### ISSUE TITLE
PY3 compatibility issue in scrapy-redis/src/scrapy_redis next_requests()

### ISSUE THREAD

**Issue by tyong920** [rolando/scrapy-redis#66]
It works fine in PY2.
but as for PY3, it sucks.

``` bash
2016-08-22 09:00:05 [scrapy] ERROR: Error caught on signal handler: <bound method RedisMixin.spider_idle of <RighticSpider 'rightic' at 0x7efe1de59390>>
Traceback (most recent call last):
  File "/home/eric/.virtualenvs/scrapy3/lib/python3.5/site-packages/scrapy/utils/signal.py", line 30, in send_catch_log
    *arguments, **named)
  File "/home/eric/.virtualenvs/scrapy3/lib/python3.5/site-packages/pydispatch/robustapply.py", line 55, in robustApply
    return receiver(*arguments, **named)
  File "/home/eric/.virtualenvs/scrapy3/lib/python3.5/site-packages/scrapy_redis/spiders.py", line 109, in spider_idle
    self.schedule_next_requests()
  File "/home/eric/.virtualenvs/scrapy3/lib/python3.5/site-packages/scrapy_redis/spiders.py", line 103, in schedule_next_requests
    for req in self.next_requests():
  File "/home/eric/.virtualenvs/scrapy3/lib/python3.5/site-packages/scrapy_redis/spiders.py", line 84, in next_requests
    req = self.make_request_from_data(data)
  File "/home/eric/.virtualenvs/scrapy3/lib/python3.5/site-packages/scrapy_redis/spiders.py", line 96, in make_request_from_data
    if '://' in data:
TypeError: a bytes-like object is required, not 'str'
2016-08-22 09:00:05 [scrapy] INFO: Closing spider (finished)
2016-08-22 09:00:05 [scrapy] INFO: Dumping Scrapy stats:
{'finish_reason': 'finished',
 'finish_time': datetime.datetime(2016, 8, 22, 1, 0, 5, 210158),
 'log_count/ERROR': 2,
 'log_count/INFO': 8,
 'start_time': datetime.datetime(2016, 8, 22, 1, 0, 4, 378904)}
2016-08-22 09:00:05 [scrapy] INFO: Spider closed (finished)
```

Consider to add decode() like this:

``` python
def next_requests(self):
        """Returns a request to be scheduled or none."""
        use_set = self.settings.getbool('REDIS_START_URLS_AS_SET')
        fetch_one = self.server.spop if use_set else self.server.lpop
        # XXX: Do we need to use a timeout here?
        found = 0
        while found < self.redis_batch_size:
            data = fetch_one(self.redis_key).decode()
            if not data:
                # Queue empty.
                break
            req = self.make_request_from_data(data)
            if req:
                yield req
                found += 1
            else:
                self.logger.debug("Request not made from data: %r", data)

        if found:
            self.logger.debug("Read %s requests from '%s'", found, self.redis_key)
```

**Comment by rafaelcapucho** [comment]
Related with #49

**Comment by kingname** [comment]
@tyong920 you should not decode directly. While the spider start to run, if the start_urls is empty, here 
```
fetch_one(self.redis_key) 
```
will be None. So it will raise error if you decode a None.

**Comment by kingname** [comment]
@tyong920 My solution is [#70](https://github.com/rolando/scrapy-redis/issues/70), I change the position of `data.decode()`

**Comment by rolando** [comment]
Related: https://github.com/rolando/scrapy-redis/pull/71

**Comment by tyong920** [comment]
@kingname It is better. Good job! Thanks!

**Comment by rolando** [comment]
Fixed in #71
